tests/misc_test.go

```diff
@@ -535,3 +535,53 @@ func TestTrivialSwitch(t *testing.T) {
 	}
 	t.Fail()
 }
+
+func TestTupleFnReturnImplicitCast(t *testing.T) {
+	var ycalled int = 0
+	x := func(fn func() (int, error)) (interface{}, error) {
+		return fn()
+	}
+	y, _ := x(func() (int, error) {
+		ycalled++
+		return 14, nil
+	})
+	if y != 14 || ycalled != 1 {
+		t.Fail()
+	}
+}
+
+var tuple2called = 0
+
+func tuple1() (interface{}, error) {
+	return tuple2()
+}
+func tuple2() (int, error) {
+	tuple2called++
+	return 14, nil
+}
+func TestTupleReturnImplicitCast(t *testing.T) {
+	x, _ := tuple1()
+	if x != 14 || tuple2called != 1 {
+		t.Fail()
+	}
+}
+
+func TestDeferNamedTupleReturnImplicitCast(t *testing.T) {
```

Comment:
Is this `TestDeferNamedTupleReturnImplicitCast` test here "just in case" of future regressions?

I ask, because this test is currently passing on `master`, without the fix applied in `compiler/statements.go`. I just want to confirm that was intentional.